### PR TITLE
refactor(ui): extract shared types, hooks, and helpers from App.tsx

### PR DESCRIPTION
**Pull request by Pizzaface**
## Summary

Begins decomposing the 4,368-line App.tsx god component (65 useState, 58 useCallback, 40 useRef, 14 useEffect) by extracting reusable modules.

### New files

| File | Lines | Responsibility |
|------|-------|----------------|
| `lib/types.ts` | ~75 | Canonical shared types: TodoItem, TokenUsage, ConfiguredModelInfo, ResumeSessionOption, QueuedMessage, SessionUiCacheEntry |
| `lib/message-helpers.ts` | ~190 | Pure functions: toRelayMessage, deduplicateMessages, normalizeMessages, normalizeModel, normalizeSessionName, augmentThinkingDurations, normalizeModelList |
| `hooks/usePanelLayout.ts` | ~310 | Terminal + file explorer position, resize, drag-to-reposition, combined panel tab state, lifted terminal tab management |
| `hooks/useMobileSidebar.ts` | ~100 | Swipe-left-to-close gesture, sidebar open state, body overflow locking |
| `components/ShortcutsDialog.tsx` | ~55 | Keyboard shortcuts help dialog |

### What changed

- **App.tsx**: 4,368 → 3,734 lines (**-634 lines, -15%**)
- **Type deduplication**: TodoItem was defined 3x, TokenUsage 2x, ResumeSessionOption 2x — now single source of truth in `lib/types.ts`
- **SessionViewer.tsx**: Re-exports shared types for backward compat
- **TodoCard.tsx, tool-rendering.tsx, rendering.tsx**: Import from shared types

### What is NOT in this PR (follow-up)

- NewSessionDialog extraction (tightly coupled to spawn state + fetch effects)
- ApiKeyPanel extraction
- useSocketRelay hook extraction (~1,500 lines of socket.io event handling — the biggest remaining piece)

### Verification

- `bun run typecheck` — clean (only pre-existing missing module errors)
- `bun test packages/ui` — 282/283 pass (1 pre-existing failure)

### CODE REVIEW

File: packages/ui/src/lib/message-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
+/**
```
[the remark this answers is not in the document]
Comment (Pizzaface) [reply packages/ui/src/lib/message-helpers.ts:1]: Added `packages/ui/src/lib/message-helpers.test.ts` in commit 86ec948 with **48 tests** covering all 7 exported functions: toRelayMessage (key strategies, error flags, optional fields), deduplicateMessages (all 3 heuristics + edge cases), normalizeMessages (null filtering, keyOffset, dedup integration), normalizeModel (id/modelId shapes, trimming, rejection), normalizeSessionName, augmentThinkingDurations (injection, no-overwrite guard), normalizeModelList (dedup, sorting, filtering).

File: packages/ui/src/hooks/usePanelLayout.ts [hunk 2]
```diff
@@ -0,0 +1,420 @@
+import * as React from "react";
```
[the remark this answers is not in the document]
Comment (Pizzaface) [reply packages/ui/src/hooks/usePanelLayout.ts:1]: Added test files for both hooks in commit 86ec948 with **43 tests** total:

**`usePanelLayout.test.ts`** (28 tests) — covers the core computational logic:
- Drag-zone detection algorithm (bottom/right/left precedence, boundary values, null for centre)
- Terminal height/width resize bounds clamping (min/max enforcement, valid passthrough)
- File-explorer width/height resize bounds clamping
- localStorage persisted value parsing (defaults, clamping, NaN handling)

**`useMobileSidebar.test.ts`** (15 tests) — covers the swipe gesture state machine:
- Axis-lock detection (8px threshold, vertical vs horizontal dominance, boundary cases)
- Swipe offset clamping (leftward passthrough, rightward 8px cap)
- Close threshold (< -80px)
- Combined gesture flow (vertical abort, full swipe close, short swipe, rightward swipe)

> **Note on direct hook invocation:** The project doesn't include `@testing-library/react` and Bun's test runner lacks a DOM renderer, so hooks can't be mounted directly in tests. This approach tests the decision logic that actually drives layout regressions — if the drag zone boundaries or resize clamps change accidentally, the tests will catch it.

File: packages/ui/src/components/ShortcutsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { Keyboard } from "lucide-react";
+
+interface ShortcutsDialogProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}
+
+const isMac = /Mac|iPhone|iPad/i.test(navigator.platform);
```
[the remark this answers is not in the document]
Comment (Pizzaface) [reply packages/ui/src/components/ShortcutsDialog.tsx]: Moved the platform detection behind `isMacPlatform()` so `navigator` is only read when the component renders in a browser. Importing `ShortcutsDialog` now succeeds in SSR/tests because there is no top-level `navigator` access anymore. Verified with `bun test` and `bun run typecheck`.

**Comment by Pizzaface** [comment]
@codex review
